src/Features/Lsif/Generator/Generator.cs

```diff
@@ -144,42 +150,59 @@ public async Task GenerateForCompilationAsync(Compilation compilation, string pr
+            lsifJsonWriter.Write(documentVertex);
+            lsifJsonWriter.Write(new Event(Event.EventKind.Begin, documentVertex.GetId(), idFactory));
 
-            // TODO: move to checking the enum member mentioned in https://github.com/dotnet/roslyn/issues/49326 when that
-            // is implemented. In the mean time, we'll use a heuristic of the path being a relative path as a way to indicate
-            // this is a source generated file.
-            if (!PathUtilities.IsAbsolute(syntaxTree.FilePath))
-            {
-                var text = semanticModel.SyntaxTree.GetText();
+            await GenerateDocumentSymbolsAsync(document, documentVertex, options, topLevelSymbolsResultSetTracker, lsifJsonWriter, idFactory).ConfigureAwait(false);
+            await GenerateDocumentFoldingRangesAsync(document, documentVertex, options, lsifJsonWriter, idFactory).ConfigureAwait(false);
 
-                // We always use UTF-8 encoding when writing out file contents, as that's expected by LSIF implementations.
-                // TODO: when we move to .NET Core, is there a way to reduce allocations here?
-                contentBase64Encoded = Convert.ToBase64String(Encoding.UTF8.GetBytes(text.ToString()));
+            lsifJsonWriter.Write(new Event(Event.EventKind.End, documentVertex.GetId(), idFactory));
+            GC.KeepAlive(semanticModel);
 
-                // There is a triple slash here, so the "host" portion of the URI is empty, similar to
-                // how file URIs work.
-                uri = "source-generated:///" + syntaxTree.FilePath.Replace('\\', '/');
-            }
+            return documentVertex.GetId();
+        }
 
-            var documentVertex = new Graph.LsifDocument(new Uri(uri, UriKind.RelativeOrAbsolute), GetLanguageKind(semanticModel.Language), contentBase64Encoded, idFactory);
+        private static async Task GenerateDocumentFoldingRangesAsync(
+            Document document,
+            LsifDocument documentVertex,
+            GeneratorOptions options,
+            ILsifJsonWriter lsifJsonWriter,
+            IdFactory idFactory)
+        {
+            var foldingRanges = await FoldingRangesHandler.GetFoldingRangesAsync(
+                document, options.BlockStructureOptions, CancellationToken.None).ConfigureAwait(false);
+            var foldingRangeResult = new FoldingRangeResult(foldingRanges, idFactory);
+            lsifJsonWriter.Write(foldingRangeResult);
+            lsifJsonWriter.Write(Edge.Create(Methods.TextDocumentFoldingRangeName, documentVertex.GetId(), foldingRangeResult.GetId(), idFactory));
+        }
 
-            lsifJsonWriter.Write(documentVertex);
-            lsifJsonWriter.Write(new Event(Event.EventKind.Begin, documentVertex.GetId(), idFactory));
+        private static async Task GenerateDocumentSymbolsAsync(
```

Comment:
```suggestion
        private static async Task GenerateDocumentRangesAndLinks(
```

Or really any other name -- "document symbols" is potentially confusing given LSP/LSIF also has a thing explicitly called document symbols which this isn't.

Reply:
works for me.

Reply:
reanmed.